### PR TITLE
Bug 1887849: Fix for  new storage pool failure_domain missing

### PR DESCRIPTION
**Pull request by GowthamShanmugam**
Signed-off-by: Gowtham Shanmugasundaram <gshanmug@redhat.com>

### CODE REVIEW

File: frontend/packages/ceph-storage-plugin/src/components/block-pool/body.tsx [hunk 1]
```diff
@@ -64,6 +65,23 @@ export const BlockPoolBody = (props: BlockPoolBodyPros) => {
   const [storageCluster, storageClusterLoaded, storageClusterLoadError] = useK8sGet<
     ListKind<StorageClusterKind>
   >(OCSServiceModel, null, CEPH_STORAGE_NAMESPACE);
+  const [pools, poolsLoaded, poolsLoadedError] = useK8sGet<ListKind<StoragePoolKind>>(
```
Comment (GowthamShanmugam) [inline frontend/packages/ceph-storage-plugin/src/components/block-pool/body.tsx:68]: Getting failure domain from default pool and assigning same to new pools

File: frontend/packages/ceph-storage-plugin/src/components/block-pool/body.tsx [hunk 2]
```diff
@@ -64,6 +65,23 @@ export const BlockPoolBody = (props: BlockPoolBodyPros) => {
...
+    CephBlockPoolModel,
+    null,
+    CEPH_STORAGE_NAMESPACE,
+  );
+
+  React.useMemo(() => {
+    if (!state.failureDomain && poolsLoaded && !poolsLoadedError) {
+      const pool: StoragePoolKind = pools?.items.find((blockPoolConfig) =>
+        isDefaultPool(blockPoolConfig),
+      );
```
Comment (afreen23) [inline frontend/packages/ceph-storage-plugin/src/components/block-pool/body.tsx:78]:
```suggestion
      const pool: StoragePoolKind = pools?.items.find(isDefaultPool);
```

**Review by afreen23 — APPROVED** [review]

File: frontend/packages/ceph-storage-plugin/src/components/block-pool/body.tsx (hunk 2 above)
Comment (GowthamShanmugam) [reply frontend/packages/ceph-storage-plugin/src/components/block-pool/body.tsx:78]: Ah! you told this in earlier PR comment, I forgot, my bad :)

**Comment by GowthamShanmugam** [comment]
@afreen23 Changes done

**Review by afreen23 — APPROVED** [review]
/lgtm

**Comment by GowthamShanmugam** [comment]
#8775

**Comment by GowthamShanmugam** [comment]
@afreen23 yes pool management code is different from 4.7 code, need to fix this differently in 4.7 branch

**Pull request merged**